005-1-locations-on-stack.md

```diff
@@ -0,0 +1,590 @@
+        - 2.5.2.3 Implicit Locations
+        - 2.5.2.4 Undefined Locations
+        - 2.5.2.5 Composite Locations
+        - 2.5.2.6 Offset Operations
+    - 2.5.3 Control Flow Operations
+    - 2.5.4 Type Conversions
+    - 2.5.5 Special Operations
+- 2.6 Expression Lists
+
+
+
+Proposed Changes
+----------------
+
+### Section 2.2 Attribute Types
+
+[Page 24]
+In Table 2.3, Classes of attribute value, in the row for "locdesc,"
+replace the reference to Section 2.6 with a reference to Section 2.5.
+
+
+### Section 2.5 DWARF Expressions
+
+[Page 26]
+Replace the contents of the preamble to this section with:
+
+> DWARF expressions describe how to compute a value or specify a
+> location. They are expressed in terms of DWARF operations that
+> operate on a stack. Each element on the stack may be
+> either a value or a location.
+> 
+> Values on the stack are typed, and can represent a value of any
+> supported base type of the target machine, or of the generic type,
+> which is an integral type that has the size of an address on the
+> target machine and unspecified signedness.
+> 
+> _The generic type is the same as the unspecified type used for stack
+> operations defined in DWARF Version 4 and before._
+> 
+> Locations represent a place where a value is stored, which may
+> be any of the following:
+> 
+> - Memory locations, for values that are stored in memory, starting
+>   at a given address within a given address space.
+> 
+> - Register locations, for values that are stored in registers.
+> 
+> - Undefined locations, for values that are unavailable
+>   (perhaps due to optimization).
+> 
+> - Implicit locations, for values that are not stored anywhere (like
+>   undefined locations), but whose values can be reconstructed.
+> 
+> - Composite locations, for values that are stored in a combination
+>   of storage (e.g., aggregate values where certain components have
+>   been promoted to registers).
+> 
+> A DWARF expression is encoded as a stream of operations, each consisting of an
+> opcode followed by zero or more literal operands. The number of operands is
+> implied by the opcode.
+> 
+> The result of a DWARF expression is the value or location on the top
+> of the stack after evaluating the operations.
+> 
+> A DWARF expression is evaluated in a context that determines whether
+> its result is expected to be a value or a location. Expressions that are
+> expected to produce a location are called "location descriptions."
+> 
+> Implicit conversions between memory addresses and values may happen
+> during the execution of any operation or when evaluation of the
+> expression is completed. If a location is expected, but the result is
+> a value, the value is implicitly treated as a memory address in the
+> default address space, and converted to a memory location. If a value
+> is expected, but the result is a memory location in the default
+> address space, the address is implicitly converted to a value.
```

Comment:
Given that values are typed, is it worth specifying here something like "... converted to a value of generic type."?